iree/samples/variables_and_state/test.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This script runs the steps laid out in the README for this sample. It is
+# intended for use on continuous integration servers and as a reference for
+# users, but can also be run manually.
+
+set -x
+set -e
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+BUILD_DIR=${ROOT_DIR}/build
+ARTIFACTS_DIR=/tmp/iree/colab_artifacts
+
+# 1. Run the notebook to generate `counter.mlir` and `counter_vmvx.vmfb`
+${ROOT_DIR}/build_tools/testing/run_python_notebook.sh \
+  ${ROOT_DIR}/iree/samples/variables_and_state/variables_and_state.ipynb
+test -f ${ARTIFACTS_DIR}/counter.mlir && echo "counter.mlir exists"
+test -f ${ARTIFACTS_DIR}/counter_vmvx.vmfb && echo "counter_vmvx.vmfb exists"
+
+# 2. Compile the `iree_samples_variables_and_state` CMake target.
+git submodule update --init
```

Comment:
This seems like the wrong level at which to be running this. Something outside of this test script should ensure submodules are up to date. That's an environment concern, not a concern of the test. It would be totally reasonable to want to test with modified submodules, for instance.

Reply:
Good point.

The Buildkite pipeline also needs a bit of rework to run multiple tests (https://github.com/google/iree/pull/6191 will be adding another sample with the same pattern):
` - "docker run --user=$(id -u):$(id -g) --volume=\\$PWD:\\$IREE_DOCKER_WORKDIR --workdir=\\$IREE_DOCKER_WORKDIR --rm gcr.io/iree-oss/samples@sha256:a34511d6387fe747a91446e19ca8913fbee49b5e15036e685fd74562803778ed iree/samples/variables_and_state/test.sh"`

I could either add another script (`iree/samples/test.sh`) that does setup and runs both, or fold that all into the pipeline. WDYT?

Reply:
Having a single docker run invocation makes sense to me. Spinning a container multiple times is silly. So I think that points to encapsulating whatever it is you want to run into a single script.

Reply:
Added a script for the second sample using this style and added a new script that inits submodules and calls both test scripts.